Apigee/Mint/Base/BaseObject.php

```diff
@@ -150,16 +150,25 @@ public function save($save_method = 'auto')
             throw new ParameterException('Valid save methods are create, update or auto.');
         }
 
-        if (!isset($this->{$this->idField})) {
```

Comment:
Because $id is usually a private property, parent class was unable to access it, therefore `$this->{$this->idField}` always returned `null`. This issue can be eliminated two ways, a) enforce definition of a _public_ `getId()` method by all subclasses and call it here every time or with the reflection trick, where we do not enforce the `getId()` method, because not all entities has `idField` set, but if it has then the idField must have a public getter method. This allows us to use `name` as an id and call `getName()` to get its value.